### DISCUSSION TITLE
Markdown のシンタックスハイライトの不具合

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#357] [要望・アイデア・不具合]
コードブロックで Markdown のシンタックスハイライトを当てると、見出しがコメントアウトや本文と同色になっています。

また、Rouge のデフォルトでは見出しレベルごとに配色が異なるようですが、統一しても良いように思います。

> [Rouge によるデモ][demo]

[demo]:
http://rouge.jneen.net/v4.0.0/markdown/IyBIZWFkaW5nIDEKClRoaXMgaXMgcGFyYWdyYXBoLgoKIyMgSGVhZGluZyAyCgojIyMgSGVhZGluZyAzCgojIyMjIEhlYWRpbmcgNAoKIyMjIyMgSGVhZGluZyA1CgojIyMjIyMgSGVhZGluZyA2Cgo8IS0tIGNvbW1lbnRvdXQgLS0-CgpIZWFkaW5nIDEKPT09CgpUaGlzIGlzIHBhcmFncmFwaC4KCkhlYWRpbmcgMgotLS0KCjwhLS0gY29tbWVudG91dCAtLT4K

### ATX headings

ATX 見出しでは、
- 見出しレベル 1 の見出し と 本文
- 見出しレベル 2 以降の見出し と コメントアウト

が同色になってしまします。

![image](https://user-images.githubusercontent.com/75878492/201744533-e9b328bf-328f-4670-a393-f07722dfb90f.png)

<details><summary>サンプルコード</summary>

````markdown
```markdown
# Heading 1

This is paragraph.

## Heading 2

### Heading 3

#### Heading 4

##### Heading 5

###### Heading 6

<!-- commentout -->
```
````

</details>

### Setext headings

Setext 見出しでは、
- 見出しレベル 1 の見出し と コメントアウト
- 見出しレベル 2 の見出し と 本文

が同色になってしまします。

![image](https://user-images.githubusercontent.com/75878492/201744226-fa4733b1-752d-4ef9-bc4c-33c820625bc2.png)

<details><summary>サンプルコード</summary>

````markdown
```markdown
Heading 1
===

This is paragraph.

Heading 2
---

<!-- commentout -->
```
````

</details>

**Comment by tanaka-asami415** [comment]
@Yarakashi-Kikohshi 
ありがとうございます！
Rougeの仕様によるものかもしれないのですが、#354 #356 と合わせて調査いたします🙇
